File: app/src/main/java/com/qadam/lastpuff/ui/screens/body/BodyScreen.kt
package com.qadam.lastpuff.ui.screens.body

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.qadam.lastpuff.domain.model.OrganInfo
import com.qadam.lastpuff.ui.components.BodySilhouette

private val BodyBackground = Color(0xFF0A0A0A)
private val CardDark = Color(0xFF1A1F1C)

@Composable
fun BodyScreen(
    organs: List<OrganInfo>,
    willpower: Float,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .background(BodyBackground)
            .verticalScroll(rememberScrollState())
            .padding(20.dp)
    ) {
        Text(
            text = "Твоё тело",
            style = MaterialTheme.typography.headlineMedium,
            color = Color.White,
            fontWeight = FontWeight.Bold
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = "Твоё тело восстанавливается каждый день без сигарет",
            style = MaterialTheme.typography.bodyMedium,
            color = Color.White.copy(alpha = 0.6f)
        )
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            text = "Индекс восстановления",
            style = MaterialTheme.typography.labelLarge,
            color = Color(0xFF4CAF50)
        )

        Spacer(modifier = Modifier.height(16.dp))

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(260.dp)
                .clip(RoundedCornerShape(16.dp))
                .background(Color(0xFF0D1520))
        ) {
            BodySilhouette(organs = organs)
        }

        Spacer(modifier = Modifier.height(16.dp))

        organs.forEach { organ ->
            OrganRecoveryCard(organ = organ)
            Spacer(modifier = Modifier.height(8.dp))
        }

        WillpowerCard(willpower = willpower)

        Spacer(modifier = Modifier.height(12.dp))

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(16.dp))
                .background(CardDark)
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text("🌿", fontSize = 28.sp)
            Spacer(modifier = Modifier.width(12.dp))
            Text(
                text = "Ты уже на пути к новой жизни! Продолжай в том же духе.",
                style = MaterialTheme.typography.bodyMedium,
                color = Color.White.copy(alpha = 0.85f)
            )
        }
    }
}

@Composable
private fun OrganRecoveryCard(organ: OrganInfo) {
    val animatedProgress by animateFloatAsState(
        targetValue = organ.value / 100f,
        animationSpec = tween(800),
        label = "organProgress"
    )

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(CardDark)
            .padding(horizontal = 14.dp, vertical = 12.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(organ.emoji, fontSize = 18.sp)
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = "${organ.label} ${organ.value.toInt()}%",
                style = MaterialTheme.typography.titleSmall,
                color = Color.White,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis
            )
        }
        Spacer(modifier = Modifier.height(8.dp))
        LinearProgressIndicator(
            progress = { animatedProgress },
            modifier = Modifier
                .fillMaxWidth()
                .height(5.dp)
                .clip(RoundedCornerShape(3.dp)),
            color = Color(organ.glowColor),
            trackColor = Color.White.copy(alpha = 0.1f)
        )
        Spacer(modifier = Modifier.height(6.dp))
        Text(
            text = organ.description,
            style = MaterialTheme.typography.bodySmall,
            color = Color.White.copy(alpha = 0.55f),
            lineHeight = MaterialTheme.typography.bodySmall.lineHeight
        )
    }
}

@Composable
private fun WillpowerCard(willpower: Float) {
    val animatedProgress by animateFloatAsState(
        targetValue = willpower / 100f,
        animationSpec = tween(800),
        label = "willpowerProgress"
    )

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(CardDark)
            .padding(16.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text("🛡", fontSize = 20.sp)
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = "Сила воли ${willpower.toInt()}%",
                style = MaterialTheme.typography.titleSmall,
                color = Color(0xFFFFD54F),
                fontWeight = FontWeight.Bold
            )
        }
        Spacer(modifier = Modifier.height(10.dp))
        LinearProgressIndicator(
            progress = { animatedProgress },
            modifier = Modifier
                .fillMaxWidth()
                .height(6.dp)
                .clip(RoundedCornerShape(3.dp)),
            color = Color(0xFFFFD54F),
            trackColor = Color.White.copy(alpha = 0.1f)
        )
    }
}

File: ios/QadamLastPuff/Views/Screens/BodyScreen.swift
import SwiftUI

struct BodyScreen: View {
    let organs: [OrganInfo]
    let willpower: Float

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                Text("Твоё тело")
                    .font(.title.bold())
                    .foregroundStyle(.white)
                Text("Твоё тело восстанавливается каждый день без сигарет")
                    .foregroundStyle(.white.opacity(0.6))
                Text("Индекс восстановления")
                    .font(.subheadline.bold())
                    .foregroundStyle(QadamColors.recoveryGreen)

                BodySilhouette(organs: organs)
                    .frame(height: 260)
                    .background(Color(hex: 0x0D1520))
                    .clipShape(RoundedRectangle(cornerRadius: 16))

                ForEach(organs) { organ in
                    OrganRecoveryCard(organ: organ)
                }

                WillpowerCard(willpower: willpower)

                HStack(spacing: 12) {
                    Text("🌿").font(.title)
                    Text("Ты уже на пути к новой жизни! Продолжай в том же духе.")
                        .foregroundStyle(.white.opacity(0.85))
                }
                .padding(16)
                .background(QadamColors.bodyCard)
                .clipShape(RoundedRectangle(cornerRadius: 16))
            }
            .padding(20)
        }
        .background(QadamColors.bodyBackground)
    }
}

private struct OrganRecoveryCard: View {
    let organ: OrganInfo
    @State private var animated: CGFloat = 0

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(organ.emoji)
                Text("\(organ.label) \(Int(organ.value))%")
                    .font(.subheadline.bold())
                    .foregroundStyle(.white)
            }
            ProgressView(value: animated)
                .tint(Color(hex: organ.glowColorHex))
            Text(organ.description)
                .font(.caption)
                .foregroundStyle(.white.opacity(0.55))
        }
        .padding(14)
        .background(QadamColors.bodyCard)
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .onAppear {
            withAnimation(.easeOut(duration: 0.8)) {
                animated = CGFloat(organ.value / 100)
            }
        }
    }
}

private struct WillpowerCard: View {
    let willpower: Float
    @State private var animated: CGFloat = 0

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack {
                Text("🛡")
                Text("Сила воли \(Int(willpower))%")
                    .font(.subheadline.bold())
                    .foregroundStyle(QadamColors.willpowerYellow)
            }
            ProgressView(value: animated)
                .tint(QadamColors.willpowerYellow)
        }
        .padding(16)
        .background(QadamColors.bodyCard)
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .onAppear {
            withAnimation(.easeOut(duration: 0.8)) {
                animated = CGFloat(willpower / 100)
            }
        }
    }
}
